test/functional/JIT_Test/src/jit/test/recognizedMethod/TestMathUtils.java

```diff
@@ -0,0 +1,311 @@
+            {fsNaN, +0.0f, fsNaNBits},
+            {-0.0f, fqNaN, fqNaNBits},
+            {fqNaN, -0.0f, fqNaNBits},
+            {-0.0f, fsNaN, fsNaNBits},
+            {fsNaN, -0.0f, fsNaNBits},
+
+            {dqNaN, dqNaN, dqNaNBits},
+            {dsNaN, dsNaN, dsNaNBits},
+            {dqNaN, dsNaN, dqNaNBits},
+            {dsNaN, dqNaN, dsNaNBits},
+
+            {+0.0d, dqNaN, dqNaNBits},
+            {dqNaN, +0.0d, dqNaNBits},
+            {+0.0d, dsNaN, dsNaNBits},
+            {dsNaN, +0.0d, dsNaNBits},
+            {-0.0d, dqNaN, dqNaNBits},
+            {dqNaN, -0.0d, dqNaNBits},
+            {-0.0d, dsNaN, dsNaNBits},
+            {dsNaN, -0.0d, dsNaNBits}
+        };
+
+        Object[][] nanPairs = new Object[60][3];
+        int i = 0;
+        while (i < constNanPairs.length) {
+            nanPairs[i] = constNanPairs[i];
+            i++;
+        }
+        while (i < nanPairs.length) {
+            NaNTestPair<Float, Integer> fpair =  getFloatNaNPair();
+            NaNTestPair<Double, Long> dpair =  getDoubleNaNPair();
+            if (fpair == null) {
+                continue;
+            }
+            nanPairs[i][0] = fpair.first;
+            nanPairs[i][1] = fpair.second;
+            nanPairs[i][2] = fpair.expected;
+            i++;
+            if (dpair == null | i >= nanPairs.length){
+                continue;
+            }
+            nanPairs[i][0] = dpair.first;
+            nanPairs[i][1] = dpair.second;
+            nanPairs[i][2] = dpair.expected;
+            i++;
+        }
+        return nanPairs;
+    }
+
+    @DataProvider(name="normalNumberProvider")
+    public static Object[][] normalNumberProvider() {
+        Object [][] numList = new Object[20][2];
+        Random r = new Random();
+        for (int i = 0; i < numList.length / 2; i++) {
+            float a = r.nextFloat() * 1000000.0f;
+            float b = r.nextFloat() * 1000000.0f;
+            int sign1 = r.nextInt(2) == 0 ? -1 : 1;
+            int sign2 = r.nextInt(2) == 0 ? -1 : 1;
+            numList[i][0] = a * sign1;
+            numList[i][1] = b * sign2;
+        }
+        for (int i = numList.length / 2; i < 20; i++) {
+            double a = r.nextDouble() * 1000000000.0d;
+            double b = r.nextDouble() * 1000000000.0d;
+            int sign1 = r.nextInt(2) == 0 ? -1 : 1;
+            int sign2 = r.nextInt(2) == 0 ? -1 : 1;
+            numList[i][0] = a * sign1;
+            numList[i][1] = b * sign2;
+        }
+        return numList;
+    }
+
+    private static NaNTestPair<Double, Long> getDoubleNaNPair(){
+        Random r = new Random();
+        NaNTestPair<Double, Long> p = new NaNTestPair<Double, Long>();
+        // 0->quiet, 1->signalling
+            String message = "";
+            double farg1, farg2, farg3;
+			farg1 = 0.0f;
+			farg2 = 0.0f;
+			farg3 = 0.0f;
+            long iarg1, iarg2;
+			iarg1 = r.nextLong() >> 13; // shift right to only mantisa has 1s, and quiet bit is not set
```

Comment:
```suggestion
         double farg1 = 0.0f;
         double farg2 = 0.0f;
         double farg3 = 0.0f;
         long iarg1 = r.nextLong() >> 13; // shift right to only mantissa has 1s, and quiet bit is not set
```

Reply:
Please reword the comment so that it becomes more clear and also move the declaration and initialization of variable closer to its use.